rotkehlchen/globaldb/handler.py

```diff
@@ -2013,31 +2017,56 @@ def get_assetid_from_exchange_name(exchange: Location | None, symbol: str, defau
         return default if identifier is None else identifier[0]
 
     @staticmethod
-    def query_location_asset_mappings(
-            filter_query: 'LocationAssetMappingsFilterQuery',
+    def query_asset_mappings_by_type(
+            mapping_type: Literal['location', 'counterparty'],
+            filter_query: Union['LocationAssetMappingsFilterQuery', 'CounterpartyAssetMappingsFilterQuery'],  # noqa: E501
     ) -> tuple[list[dict[str, str | Location | None]], int, int]:
-        """Returns a tuple with the mappings, their amount according to the filter_query, and total
-        amount without any filter. Mappings are in the form of a list of dicts with the keys,
-        asset, location, and location_symbol."""
+        """Query asset mappings based on the mapping type.
+
+        Returns:
+          - A list of mapping dictionaries.
+          - The count of mappings matching the filter.
+          - The total count of mappings.
+
+        For location mappings, keys are: 'asset', 'location', 'location_symbol'.
+        For counterparty mappings, keys are: 'asset', 'counterparty', 'counterparty_symbol'.
+        """
+        if mapping_type == 'location':
+            columns = 'local_id, location, exchange_symbol'
+        else:  # counterparty
+            columns = 'local_id, counterparty, symbol'
+
         with GlobalDBHandler().conn.read_ctx() as cursor:
             mappings_total = cursor.execute(
-                'SELECT COUNT(*) FROM location_asset_mappings',
+                f'SELECT COUNT(*) FROM {mapping_type}_asset_mappings',
             ).fetchone()[0]
 
             query, bindings = filter_query.prepare(with_pagination=False)
             mappings_count = cursor.execute(
-                f'SELECT COUNT(*) FROM location_asset_mappings {query}', bindings,
+                f'SELECT COUNT(*) FROM {mapping_type}_asset_mappings {query}', bindings,
             ).fetchone()[0]
 
             query, bindings = filter_query.prepare()
             cursor.execute(
-                f'SELECT local_id, location, exchange_symbol FROM location_asset_mappings {query}', bindings,  # noqa: E501
+                f'SELECT {columns} FROM {mapping_type}_asset_mappings {query}', bindings,
             )
-            return [{
-                'asset': identifier,
-                'location': None if location is None else str(Location.deserialize_from_db(location)),  # noqa: E501
-                'location_symbol': symbol,
-            } for identifier, location, symbol in cursor], mappings_count, mappings_total
+
+            mappings = []
+            for identifier, location_or_counterparty, symbol in cursor:
+                if mapping_type == 'location':
```

Comment:
Come on ... an if in every loop iteration, and for a value that will always be the same in the entire function's lifetime? Abstract it properly